README.md

```diff
@@ -528,6 +529,38 @@ filtered. Modifiable payload includes:
 notice[:params][:my_param] = 'foobar'
 ```
 
+### Custom exception attributes
+
+The library supports custom exception attributes. This is useful if you work
+with custom exceptions, which define non-standard attributes and you can't
+attach any additional data with help of the [`add_filter`](#airbrakeadd_filter)
+API due to the fact that the data isn't available at confuration time yet.
+
+In this case, you could define a special hook method on your exception called
+`#to_airbrake`. The method must return a Hash the keys of which must be a subset
+of the ones mentioned in the [`Notice#[]`](#notice--notice) API.
+
+```ruby
+class MyException
+  def initialize
+    @http_code = 404
+  end
+
+  # The library expects you to define this method. You must return a Hash,
+  # containing the keys you want to modify.
+  def to_airbrake
```

Comment:
I understand what you want to achieve, but:

1. A Hash must be returned, so we could merge it with the Notice object, so it could be filtered & truncated, if needed
2. We could achieve what you suggest only if we allowed populating the "params" bit of the payload (ignoring context, session and the rest), but how useful would that be? I think we could afford it for the sake of simpler implementation, but we still can't overcome (1)
3. How can we detect if we should append the exception to the Notice object? Currently we check if `to_airbrake` is defined. With your proposed API this won't be possible and we would need to do `notice.params[exception] = exc` for every exception, which doesn't look good. By default, `to_json` on exception object seems to be calling `to_s`, so if the user doesn't define it, then we get a string like:
 `"params":{"exception":"\"MyException\""}` instead of
 `"params":{"exception":{"foo":"bar"}}`. Because of Rails' ActiveSupport your idea would work, but for vanilla Ruby it doesn't work like that, unfortunately.

P.S. This is how the exception would look like

```rb
class MyException < StandardError do
  def initialize(params)
    @params = params
  end
 
  def to_json(*a)
    @params.to_json(*a)
  end
end

ex = MyException.new(foo: 'bar')
```